src/peer_digest.cc

```diff
@@ -65,19 +65,10 @@ static const time_t GlobDigestReqMinGap = 1 * 60;   /* seconds */
 
 static time_t pd_last_req_time = 0; /* last call to Check */
 
-PeerDigest::PeerDigest(CachePeer * p)
+PeerDigest::PeerDigest(CachePeer *p) :
+    peer(p),
+    host(peer->host) // if peer disappears, we will know it's name
```

Comment:
If we can make the `host` member constant after this initialization improvement, let's do that.